sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -21,11 +22,14 @@
 import com.azure.cosmos.implementation.routing.Range;
 import com.azure.cosmos.models.ModelBridgeInternal;
 import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 import java.time.Instant;
 import java.util.Collections;
+import java.util.ConcurrentModificationException;
```

Comment:
does anything through this? `ConcurrentModificationException`

Reply:
LinkedHashMap could throw this

Reply:
why do we still need this?